pkg/services/app/add.go

```diff
@@ -0,0 +1,503 @@
+package app
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"github.com/pkg/errors"
+	"github.com/weaveworks/weave-gitops/pkg/git"
+	"github.com/weaveworks/weave-gitops/pkg/kube"
+)
+
+type DeploymentType string
+type SourceType string
+type ConfigType string
+
+const (
+	DeployTypeKustomize DeploymentType = "kustomize"
+	DeployTypeHelm      DeploymentType = "helm"
+
+	SourceTypeGit  SourceType = "git"
+	SourceTypeHelm SourceType = "helm"
+
+	ConfigTypeUserRepo ConfigType = ""
+	ConfigTypeNone     ConfigType = "NONE"
+)
+
+type AddParams struct {
+	Dir                  string
+	Name                 string
+	Url                  string
+	Path                 string
+	Branch               string
+	PrivateKey           string
+	PrivateKeyPass       string
+	DeploymentType       string
+	Chart                string
+	SourceType           string
+	CommitManifests      bool
+	AutomationRepo       string
+	AutomationRepoPath   string
+	AutomationRepoBranch string
+	Namespace            string
+	DryRun               bool
+}
+
+func (a *App) Add(params AddParams) error {
+	fmt.Print("Updating parameters from environment... ")
+	params, err := a.updateParametersIfNecessary(params)
+	if err != nil {
+		return errors.Wrap(err, "could not update parameters")
+	}
+
+	fmt.Print("done\n\n")
```

Comment:
We will need to come up with a plan for how to do these `fmt` calls to `stdout`. Maybe we need to add a `log.Writer` to the service deps to capture it as part of a single request.

Does not need to be solved in this PR necessarily.

Reply:
Agree, I was thinking of adding something similar to flux so we have those nice little icons in the messages, and when calling from the API we could have a more structured format for it.